library/std/src/sys/sgx/thread.rs

```diff
@@ -121,9 +121,15 @@ impl Thread {
         rtassert!(wait_error.kind() == io::ErrorKind::WouldBlock);
     }
 
-    pub fn set_name(_name: &CStr) {
-        // FIXME: could store this pointer in TLS somewhere
-    }
+    /// The concept of an operating system thread name is not applicable to SGX as SGX works
+    /// outside of concepts of an operating system.
+    /// Any thread information from the OS is not to be trusted inside the SGX security enclave.
+    ///
+    /// The internally visible SGX thread name, on the other hand, is already saved and exposed
+    /// by the standard platform-agnostic Rust Thread definition.
+    ///
+    /// Therefore, this method is intentionally No-Op.
```

Comment:
This is confusing. The enclave should indeed not trust the OS, but the thread local storage is within the enclave boundaries. So it _could_ be implemented here.
I'd remove the FIXME and state in the PR description that the thread name is already stored in thread local storage as part of the [thread info](https://github.com/rust-lang/rust/blob/fa8762b7b6c2b75d6c83fb011ee8fa4874168829/library/std/src/thread/mod.rs#L527). You could also point to the [test_named_thread](https://github.com/rust-lang/rust/blob/master/library/std/src/thread/tests.rs#LL29C4-L29C21) test to prove that things already work as expected.

Reply:
cc: @jethrogb

Reply:
> but the thread local storage is within the enclave boundaries. So it _could_ be implemented here.

That is true. However, here I'm writing the rustdoc to the `set_name` method, so saying that the method is intentionally No-Op is correct? Or in other words, which paragraph is concerning you @raoulstrackx?

Reply:
No-op is correct. Referring to the unit test seems like a good idea.

Reply:
Makes sense, I've added this explanation:

> This can be observed in the [`std::thread::tests::test_named_thread`] test,
> which succeeds as-is with the SGX target.

(ammended and pushed)